snorkel/labeling/model/utils.py

```diff
@@ -3,28 +3,6 @@
 
 import numpy as np
 import torch
-from torch.utils.data import Dataset
-
-
-class MetalDataset(Dataset):
-    """A dataset that group each item in X with its label from Y
-
-    Args:
-        X: an n-dim iterable of items
-        Y: a torch.Tensor of labels
-            This may be predicted (int) labels [n] or probabilistic (float) labels [n, k]
-    """
-
-    def __init__(self, X, Y):
-        self.X = X
-        self.Y = Y
-        assert len(X) == len(Y)
-
-    def __getitem__(self, index):
-        return tuple([self.X[index], self.Y[index]])
-
-    def __len__(self):
-        return len(self.X)
 
 
 def recursive_merge_dicts(x, y, misses="report", verbose=None):
```

Comment:
This also exists in `snorkel.classification.utils`, but the signature of LabelModel is simple enough that I think we should just move the defaults into the model definition and drop this extra copy.

Reply:
More involved that one might imagine. Leaving for now.